java/core/src/main/java/com/microsoft/bond/MapBondType.java

```diff
@@ -181,6 +187,28 @@ protected final void serializeValue(SerializationContext context, Map<TKey, TVal
     }
 
     @Override
+    protected final Map<TKey, TValue> deserializeValue(UntaggedDeserializationContext context) throws IOException {
+        final int count = context.reader.readContainerBegin();
+        Map<TKey, TValue> value = newDefaultValue(count);
```

Comment:
Please see the same comment I have for Set. I suggest (int)(count / 0.75 + 1) as initial capacity.